yt_idv/rendering_contexts/__init__.py

```diff
@@ -18,6 +16,13 @@ def render_context(engine="pyglet", **kwargs):
     RenderingContext
 
     """
+
+    # PYOPENGL_PLATFORM must be set before any opengl imports
+    if engine in ("osmesa", "egl"):
+        os.environ["PYOPENGL_PLATFORM"] = engine
```

Comment:
There's some distinction in how this sets the environment (like, not to parent process) that I think works in our favor.  So, good!